src/NzbDrone.Core/ImportLists/AniList/List/AniListParser.cs

```diff
@@ -0,0 +1,128 @@
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using NLog;
+using NzbDrone.Common.Instrumentation;
+using NzbDrone.Common.Serializer;
+using NzbDrone.Core.ImportLists.Exceptions;
+using NzbDrone.Core.ImportLists.Simkl;
+using NzbDrone.Core.MetadataSource;
+using NzbDrone.Core.Parser.Model;
+
+namespace NzbDrone.Core.ImportLists.AniList.List
+{
+    public class AniListParser : IParseImportListResponse
+    {
+        private static readonly Logger Logger = NzbDroneLogger.GetLogger(typeof(SimklParser));
+
+        private readonly AniListSettings _settings;
+        private readonly ISearchForNewSeries _seriesSearchService;
+
+        public AniListParser(AniListSettings settings, ISearchForNewSeries seriesSearchService)
+        {
+            _seriesSearchService = seriesSearchService;
+            _settings = settings;
+        }
+
+        public virtual IList<ImportListItemInfo> ParseResponse(ImportListResponse importListResponse)
+        {
+            return ParseResponse(importListResponse, out _);
+        }
+
+        public IList<ImportListItemInfo> ParseResponse(ImportListResponse importListResponse, out PageInfo pageInfo)
+        {
+            var result = new List<ImportListItemInfo>();
+
+            if (!PreProcess(importListResponse))
+            {
+                pageInfo = null;
+                return result;
+            }
+
+            var jsonResponse = STJson.Deserialize<MediaPageResponse>(importListResponse.Content);
+
+            if (jsonResponse?.Data?.Page?.MediaList == null)
+            {
+                pageInfo = null;
+                return result;
+            }
+
+            // Anilist currently does not support filtering this at the query level, they will get filtered out here.
+            var filtered = jsonResponse.Data.Page.MediaList
+                .Where(x => ValidateMediaStatus(x.Media));
+
+            foreach (var item in filtered)
+            {
+                var media = item.Media;
+                var mappedSeries = _seriesSearchService.SearchForNewSeriesByAniListId(media.Id.ToString())
```

Comment:
Instead of mapping during the processing results should be returned with their AniListId and when all the results are processed in `ImportListSyncService` Sonarr will perform the search. This will mean Skyhook doesn't get called twice as well, since it would just get the series data when it's looking to add the series instead of once here and once when adding the series.

How it's done for IMDB: https://github.com/Sonarr/Sonarr/blob/b1d12b8ee9b828c780e88ef029f2bb8e0934f053/src/NzbDrone.Core/ImportLists/ImportListSyncService.cs#L84

Reply:
This approach is much better, thanks.